connector/connect/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectPlannerSuite.scala

```diff
@@ -31,8 +31,11 @@ import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
  * test cases.
  */
 trait SparkConnectPlanTest {
+
+  def getSession(): SparkSession = None.orNull
```

Comment:
is it possible to reuse `SharedSparkSessionBase`?

Reply:
This is relevant to the comment above: the `SharedSparkSession` and its base defines implicit which will cause ambiguity with Catalyst implicit. For current `SparkConnectProtoSuite`, we cannot let it inherit `SharedSparkSession`.  Meanwhile for the testing purpose on this `Deduplicate` implementation, we need a session. This is why this PR does some refactoring on the testing suites have a separation.